tests/Operation/ModifyCollectionFunctionalTest.php

```diff
@@ -25,7 +25,21 @@ public function testCollMod()
         );
         $result = $modifyCollection->execute($this->getPrimaryServer());
 
-        $this->assertSame(3, $result['expireAfterSeconds_old']);
-        $this->assertSame(1000, $result['expireAfterSeconds_new']);
+        if (array_key_exists('raw', $result)) {
+            /* Sharded environment, where we only assert if a shard had a successful update. For
+             * non-primary shards that don't have chunks for the collection, the result contains a
+             * "ns does not exist" error. */
+            foreach ($result['raw'] as $shard) {
+                $shard = (array) $shard;
```

Comment:
Why not add `'document' => 'array'` to the ModifyCollection `'typeMap` option to avoid this cast? We're already doing it for "root" to allow assertions on `$result['expireAfterSeconds_old']` and the like.